gamemodes/terrortown/gamemode/shared/sh_player_ext.lua

```diff
@@ -134,6 +132,10 @@ function plymeta:SetRole(subrole, team, forceHooks)
 		hook.Run("TTT2UpdateTeam", self, oldTeam, newTeam)
 	end
 
+	if SERVER and not suppressEvent and (oldRole ~= subrole or oldTeam ~= newTeam or forceHooks) then
```

Comment:
inside the event is a check that discards all events where the subrole or team is `nil`
Changes to/from `ROLE_NONE` are added though. I think it is okay, because some addons might want to use these events